crates/subspace-farmer/src/plot.rs

```diff
@@ -385,6 +390,46 @@ impl Plot {
         })?
     }
 
+    /// Returns sequential piece indexes for piece retrieval
+    pub(crate) fn read_sequential_piece_indexes(
+        &self,
+        from_index_hash: PieceIndexHash,
+        count: u64,
+    ) -> io::Result<Vec<PieceIndex>> {
+        let (result_sender, result_receiver) = mpsc::channel();
+
+        self.inner
+            .requests_sender
+            .send(RequestWithPriority {
+                request: Request::ReadPieceIndexes {
+                    from_index_hash,
+                    count,
+                    result_sender,
+                },
+                priority: RequestPriority::High,
+            })
+            .map_err(|error| {
+                io::Error::other(format!(
+                    "Failed sending read piece indexes request: {error}"
+                ))
+            })?;
+
+        result_receiver.recv().map_err(|error| {
+            io::Error::other(format!(
+                "Read piece indexes result sender was dropped: {error}",
+            ))
+        })?
+    }
+
+    #[allow(dead_code)]
```

Comment:
Why do we need an iterator at all? We can just return up to `N` elements instead. Seems like overengineering to me again.

Reply:
I thought that it would lead to less inconsistency and errors. Also, we can increase/decrease batching of retrieval of piece indexes, which would be helpful.

Reply:
It makes more sense to me to provide max number of pieces to read as an argument and just read all the offsets up until such limit all with one query.